content/en/blog/_posts/clientcmd-tutorial.md

```diff
@@ -0,0 +1,275 @@
+```
+
+### Bind the flags
+
+Once a set of flags has been defined,
+it can be used to bind command-line flags to overrides using
+[`clientcmd.BindOverrideFlags`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#BindOverrideFlags).
+This requires a
+[`pflag`](https://pkg.go.dev/github.com/spf13/pflag) `FlagSet`
+rather than one from Go’s flag package.
+
+If you also want to bind `--kubeconfig`, you should do so now,
+by binding `ExplicitPath` in the loading rules:
+
+```lang-go
+flags.StringVarP(&loadingRules.ExplicitPath, "kubeconfig", "", "", "absolute path(s) to the kubeconfig file(s)")
+```
+
+### Build the merged configuration
+
+Two functions are available to build a merged configuration:
+
+* [`clientcmd.NewInteractiveDeferredLoadingClientConfig`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#NewInteractiveDeferredLoadingClientConfig)
+* [`clientcmd.NewNonInteractiveDeferredLoadingClientConfig`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#NewNonInteractiveDeferredLoadingClientConfig)
+
+As the names suggest, the difference between the two is that the first
+can ask for authentication information interactively,
+using a provided reader,
+whereas the second only operates on the information given to it by the caller.
+
+The “deferred” mention in these function names refers to the fact that
+the final configuration will be determined as late as possible.
+This means that these functions can be called before the command-line flags are parsed,
+and the resulting configuration will use whatever values have been parsed
+by the time it’s actually constructed.
+
+### Obtain an API client
+
+The merged configuration is returned as a
+[`ClientConfig`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#ClientConfig) instance.
+An API client can be obtained from that by calling the `ClientConfig()` method.
+
+If no configuration is given
+(`KUBECONFIG` is empty or points to non-existent files,
+`~/.kube/config` doesn’t exist,
+and no configuration is given using command-line flags),
+the default setup will return an obscure error referring to `KUBERNETES_MASTER`.
+This is legacy behaviour;
+several attempts have been made to get rid of it,
+but it is preserved for the `--local` and `--dry-run` flags in `--kubectl`.
+You should check for “empty configuration” errors by calling `clientcmd.IsEmptyConfig()`
+and provide a more explicit error message.
+
+The `Namespace()` method is also useful:
+it returns the namespace that should be used.
+It also indicates whether the namespace was overridden by the user
+(using `--namespace`).
+
+## Full example
+
+Here’s a complete example.
+
+```lang-go
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/pflag"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func main() {
+	// Loading rules, no configuration
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+
+	// Overrides and flag setup
```

Comment:
Here's one way to reconcile `github.com/spf13/pflag` with the jargon from the wider world of IT.
```suggestion
	// Overrides and flag (command line option) setup
```